### PR TITLE
Compose Poll creation UI attachment components (1)

### PR DESCRIPTION
**Pull request by skydoves**
### 🎯 Goal

[WIP] Compose Poll creation UI components

### 🛠 Implementation details

- [x] Add the poll icon on the attachment bottom sheet 
- [x] Navigate to the poll creation screen with a transition animation
- [x] Navigation back to the chat screen with a transition animation
- [x] The entire poll creation screen (questions with input fields, options with input fields, and they should be 
- [x] Nested vertical scrollable (dynamic height of a list) 
- [x] movable/reordering by drag and drop, multiple switches to limit poll options)
- [ ] Implement an AlertDialog to ask whether a user wants to discard the poll creation screen.

### 🎨 UI Changes

Screenshot 2024-06-25 at 6 56 25 PM

Screenshot 2024-06-26 at 3 36 08 PM

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Review by aleksandar-apostolov — APPROVED** [review]

File: docusaurus/android_versioned_docs/version-5.x.x/04-compose/05-message-components/05-attachments-picker.mdx [hunk 1]
```diff
@@ -147,6 +147,7 @@ class AttachmentsPickerCustomTabFactory: AttachmentsPickerTabFactory {
 
     @Composable
     override fun pickerTabContent(
+        onAttachmentPickerAction: (AttachmentPickerAction) -> Unit,
```
Comment (kanat) [inline docusaurus/android_versioned_docs/version-5.x.x/04-compose/05-message-components/05-attachments-picker.mdx:150]: Should we change the docs for v5 if this feature will be available in v6 only?

File: docusaurus/android_versioned_docs/version-6.x.x-before-refactoring/05-compose/05-message-components/05-attachments-picker.mdx [hunk 2]
```diff
@@ -161,6 +161,7 @@ class AttachmentsPickerCustomTabFactory: AttachmentsPickerTabFactory {
 
     @Composable
     override fun PickerTabContent(
+        onAttachmentPickerAction: (AttachmentPickerAction) -> Unit,
```
Comment (kanat) [inline docusaurus/android_versioned_docs/version-6.x.x-before-refactoring/05-compose/05-message-components/05-attachments-picker.mdx:164]: Same here. it's v6 before refactoring

**Review by kanat — APPROVED** [review]
LGTM

**Comment by kanat** [comment]
@skydoves I can see in the PR summary
```
[WIP] Compose Poll creation UI components
```

Is this still in WIP?

**Comment by kanat** [comment]
@skydoves which PR should merged first, this one or https://github.com/GetStream/stream-chat-android/pull/5308? 🙂

File: docusaurus/android_versioned_docs/version-5.x.x/04-compose/05-message-components/05-attachments-picker.mdx (hunk 1 above)
Comment (skydoves) [reply docusaurus/android_versioned_docs/version-5.x.x/04-compose/05-message-components/05-attachments-picker.mdx]: That's a great point! Fixed: [57354ef](https://github.com/GetStream/stream-chat-android/pull/5305/commits/57354efde216db50a3fc4a4ec85cce83f73fb0ec)

File: docusaurus/android_versioned_docs/version-6.x.x-before-refactoring/05-compose/05-message-components/05-attachments-picker.mdx (hunk 2 above)
Comment (skydoves) [reply docusaurus/android_versioned_docs/version-6.x.x-before-refactoring/05-compose/05-message-components/05-attachments-picker.mdx]: Fixed: [57354ef](https://github.com/GetStream/stream-chat-android/pull/5305/commits/57354efde216db50a3fc4a4ec85cce83f73fb0ec)

**Comment by skydoves** [comment]
@kanat This feature is quite large, so I split it into three different PRs. We should merge them in reverse order: (3 #5311) -> (2 #5308) -> (1 #5305).

**Pull request merged**